go/luhn/luhn.go

```diff
@@ -0,0 +1,43 @@
+//This package implements the Luhn checksum in order to validate a
+// numeric string.
+package luhn
+
+import "regexp"
+
+/*  Valid implements the Luhn checksum function. Given the target
+    string as a parameter, it returns the result of the algorithm as
+	a boolean.
+	The algorithm iterates over the string and multiplies every
+	odd digit by two, then substracts 9 to it if the result was over 9.
+	The sum of all these digits is computed, the input is considered
+	valid if said sum is divisible by 10.  */
+func Valid(input string) bool {
+	ok := false
+	digitSum := 0
+
+	spaceStrippingRegex := regexp.MustCompile(" ")
+	input = spaceStrippingRegex.ReplaceAllString(input, "")
+	isNumeric, _ := regexp.MatchString("^[0-9]", input)
```

Comment:
are you sure that is the needed regexp? I didn't read the text of the task :blush:, but I would suppose you meant `"^[0-9]+$"`.

Reply:
I absolutely did, What was I thinking ?
I am adding a new test case using the "55>"  input string. It manages to go through the false regex and yield a true result. The fix obviously does not let it through.